components/kubernetes/component.go

```diff
@@ -27,3 +27,8 @@ type Cluster struct {
 func (c *Cluster) Export(ctx *pulumi.Context, out *ClusterOutput) error {
 	return components.Export(ctx, c, out)
 }
+
+type WorkloadComponent struct {
```

Comment:
nit: It's already a component `WorkloadComponent` -> `Workload`

Reply:
Added `Component` to make it more explicit from the caller that this is a component. We could move the file to a `components` folder and package and there it would be `componentsk8s.Workload`, but in `kubernetes` package it is not clear from the caller that this is a component, without looking into the struct implementation.

Reply:
I looked for guidance in https://go.dev/doc/effective_go, they use [both conventions ](https://go.dev/doc/effective_go#embedding)

From Go standard library I see that adding the name of the embedded struct is a practice adopted, for example in [bufio.Reader](https://cs.opensource.google/go/go/+/master:src/bufio/bufio.go;l=32-39)

`Component` is a relevant information for caller, it determines the nature of the struct: a `WorkloadComponent` is a component representing a workload. `Workload` would be too generic.